src/main/java/Bean.java

```diff
@@ -0,0 +1,147 @@
+        System.out.println(SEPARATOR);
+    }
+
+    private static void printNoValueForFields() {
+        System.out.println(ERROR_NO_VALUE_FOR_REQUIRED_FIELDS);
+        System.out.println(SEPARATOR);
+    }
+
+    private static void printNoSuchCommand() {
+        System.out.println(ERROR_NO_SUCH_COMMAND);
+        System.out.println(SEPARATOR);
+    }
+
+    private static void printGoodbyeMessage() {
+        System.out.println(MESSAGE_GOODBYE);
+        System.out.println(SEPARATOR);
+    }
+
+    private static void processAndExecute(String line, TaskList listOfTasks) {
+        Parser userLine = new Parser(line);
+        if (userLine.getCommand().equals("list")) {
+            printTaskList(listOfTasks);
+
+        } else if (userLine.getCommand().startsWith("mark")) {
+            int taskIndex = Integer.parseInt(userLine.getArgument()) - 1;
+            if (taskIndex >= 0 && taskIndex < listOfTasks.getNumTasks()) {
+                Task markedTask = listOfTasks.markTask(taskIndex, true);
+                printTaskDone(markedTask);
+            } else {
+                printInvalidTaskNo();
+            }
+
+        } else if (userLine.getCommand().startsWith("unmark")) {
+            int taskIndex = Integer.parseInt(userLine.getArgument()) - 1;
+            if (taskIndex >= 0 && taskIndex < listOfTasks.getNumTasks()) {
+                Task unmarkedTask = listOfTasks.markTask(taskIndex, false);
+                printTaskUndone(unmarkedTask);
+            } else {
+                printInvalidTaskNo();
+            }
+
+        } else if (userLine.getCommand().equals("todo")) {
+            String description = userLine.getArgument();
+            if (description.isEmpty()) {
+                printNoValueForFields();
+            } else {
+                Task newTask = listOfTasks.addTask(description);
+                printTaskAdded(newTask, listOfTasks.getNumTasks());
+            }
+
+        } else if (userLine.getCommand().equals("deadline")) {
+            String description = userLine.getArgument();
+            String by = userLine.getValue("by");
+            if (by.isEmpty() || description.isEmpty()) {
+                printNoValueForFields();
+            } else {
+                Task newTask = listOfTasks.addTask(description, by);
+                printTaskAdded(newTask, listOfTasks.getNumTasks());
+            }
+
+        } else if (userLine.getCommand().equals("event")) {
+            String description = userLine.getArgument();
+            String start = userLine.getValue("start");
+            String end = userLine.getValue("end");
+            if (start.isEmpty() || end.isEmpty() || description.isEmpty()) {
+                printNoValueForFields();
+            } else {
+                Task newTask = listOfTasks.addTask(description, start, end);
+                printTaskAdded(newTask, listOfTasks.getNumTasks());
+            }
+
+        } else {
+            printNoSuchCommand();
+        }
+    }
```

Comment:
this method had many if/else statements, with each statement having many lines of code, which can be quite hard to read. one suggestion is to make other functions that can be called within these if/else statements to make the whole thing shorter and more readable